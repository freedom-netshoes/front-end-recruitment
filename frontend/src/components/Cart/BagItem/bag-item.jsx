import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import { connect } from 'react-redux'
import CSSModules from 'react-css-modules'

import Quantity from './Quantity/quantity'
import ItemSize from './ItemSize/item-size'
import Image from '../../crossComponents/Image/image'
import Title from '../../crossComponents/Title/title'
import Price from '../../crossComponents/Price/price'
import ButtonDismiss from '../../crossComponents/ButtonDismiss/button-dismiss'
import { removeProductCart } from '../../../actions/action'
import styles from './bag-item.less'

@CSSModules(styles)
class BagItem extends Component {
  constructor (props) {
    super(props)
    this.state = {
      mouseOver: false
    }
  }
  static defaultProps = {
    product: {}
  };

  static propTypes = {
    product: PropTypes.object
  };

  mouseEnter () {
    this.setState({
      mouseOver: true
    })
  }

  mouseLeave () {
    this.setState({
      mouseOver: false
    })
  }

  render () {
    const {
      title,
      sku,
      price,
      quantity,
      availableSizes
    } = this.props.product
    return (
      <li className={`bag-item row middle-xs${this.state.mouseOver ? ' remove' : ''}`}>
        <hr />
        <div {...{ className: 'col-xs-2 start-xs', styleName: 'image-container' }}>
          <Image {...{ sku: sku, title: title, isThumb: true }} />
        </div>
        <div className='col-xs-7 start-xs'>
          <Title {...{ title: title }} />
          <ItemSize {...{ sizes: availableSizes.join(' | ') }} />
          <Quantity {...{ quantity: quantity }} />
        </div>
        <div className='col-xs-3 end-xs'>
          <ButtonDismiss {...{
            clickButton: () => {
              this.props.dispatch(removeProductCart(this.props.product))
            },
            mouseEnter: this.mouseEnter.bind(this),
            mouseLeave: this.mouseLeave.bind(this),
            extraCssClass: 'col-xs-offset-11'
          }}
          />
          <Price {...{ value: price * quantity, type: 'separated' }} />
        </div>
      </li>
    )
  }
}

export default connect()(BagItem)
