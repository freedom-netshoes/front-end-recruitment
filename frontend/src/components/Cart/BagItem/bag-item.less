@import url('../../../styles/_less-variables.less');
@import url('../../../styles/_less-mixins.less');
:global {
    #cart-shop {
        .bag-item {
            transition: opacity .2s ease-in-out;
						position: relative;
            &.remove {
                opacity: 0.2;
                h3,
                p,
                .price {
                    text-decoration: line-through;
                }
            }
            .price {
                font-size: 18px;
            }
            figure {
                height: 50px;
                margin: 0;
                width: 50px;
            }
        }
    }
}

.image-container {
    align-items: center;
    display: flex;
    height: 90px;
    justify-content: center;
}