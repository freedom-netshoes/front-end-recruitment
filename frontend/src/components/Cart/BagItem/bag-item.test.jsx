/* global describe, it, expect */
import React from 'react'
import BagItem from './bag-item'
import getStore from '../../../resources/store'
import { shallow } from 'enzyme'

describe('Testing <BagItem /> component', () => {
  it('renders the BagItem component', () => {
    const component = shallow(<BagItem store={getStore()} />)
    expect(component).toHaveLength(1)
  })
  it('Testing snapshop in BagItem ', () => {
    let testObj = {
      'id': 1,
      'sku': '18644119330491312',
      'title': 'Camisa Nike Corinthians II',
      'description': '14/15 s/nยบ',
      'availableSizes': ['S', 'G', 'GG', 'GGG'],
      'style': 'Preta com listras brancas',
      'price': 229.9,
      'installments': 9,
      'currencyId': 'BRL',
      'currencyFormat': 'R$',
      'isFreeShipping': true,
      'quantity': 2
    }
    let component = shallow(<BagItem store={getStore()} product={testObj} />)
    expect(component).toMatchSnapshot()
  })
})
