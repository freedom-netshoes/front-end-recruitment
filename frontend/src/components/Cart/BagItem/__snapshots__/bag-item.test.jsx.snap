// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <BagItem /> component Testing snapshop in BagItem  1`] = `
<BagItem
  dispatch={[Function]}
  product={
    Object {
      "availableSizes": Array [
        "S",
        "G",
        "GG",
        "GGG",
      ],
      "currencyFormat": "R$",
      "currencyId": "BRL",
      "description": "14/15 s/nÂº",
      "id": 1,
      "installments": 9,
      "isFreeShipping": true,
      "price": 229.9,
      "quantity": 2,
      "sku": "18644119330491312",
      "style": "Preta com listras brancas",
      "title": "Camisa Nike Corinthians II",
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
